File: Sorting/insertion_sort/insertionSort.js
/**
 * @arr array of integers
 */
function InsertionSort(arr) {
  if (arr.length <= 1) {
    return;
  }
  for (let i = 1; i < arr.length; ++i) {
    let value = arr[i];
    let j = i - 1;
    for (; j >= 0; --j) {
      if (arr[j] > value) {
        // move the element one position behind
        arr[j + 1] = arr[j];
      } else {
        break;
      }
    }
    arr[j + 1] = value;
  }
}

module.exports = InsertionSort;

File: Sorting/insertion_sort/python/insertionSort.py
class InsertionSort:
    def insertionSort(self, arr):
        if len(arr) <= 1:
            return arr
        for i in range(1, len(arr)):
            val = arr[i]
            j = i - 1
            while j >= 0:
                if arr[j] > val:
                    arr[j+1] = arr[j]
                else:
                    break
                j -= 1
            arr[j+1] = val
        return arr
